cli/script/command-executor.ts

```diff
@@ -554,17 +557,25 @@ function loginWithExternalAuthentication(action: string, serverUrl?: string): Pr
 }
 
 function logout(command: cli.ILogoutCommand): Promise<void> {
+    var delayedError: Error;
     return Q(<void>null)
         .then((): Promise<void> => {
-            if (!command.isLocal) {
+            if (!connectionInfo.preserveAccessKeyOnLogout) {
                 return sdk.removeAccessKey(sdk.accessKey)
                     .then((): void => {
-                        log("Removed access key " + sdk.accessKey + ".");
-                        sdk = null;
+                        log(`Removed access key ${sdk.accessKey}.`);
                     });
+            } else {
+                log("Warning: Your access key is still valid for future sessions. Please explicitly remove it if desired.");
```

Comment:
Is this warning actually necessary? I feel like the new behavior is intuitive, and therefore, I'm not sure if it makes sense to warn the user here.